jobs/sql/extractions/salud_mental_encuentro_staging.sql

```diff
@@ -1028,383 +334,91 @@ SELECT concept_id INTO @obsessive_compulsive_disorder FROM concept WHERE uuid='1
-);
-
-UPDATE salud_mental_encountero t 
-SET t.adaptive_disorders  = FALSE 
-WHERE t.adaptive_disorders  IS NULL;
-
-
--- --------------------------- Next Scheduled Appointment --------------------------------------------------------------------------
-SELECT concept_id INTO @next_appt FROM concept WHERE uuid='3ce94df0-26fe-102b-80cb-0017a47871b2';
-
-UPDATE salud_mental_encountero t 
-SET t.next_appointment  = (
-SELECT cast(value_datetime AS date)
-FROM obs o
-WHERE concept_id=@next_appt
-and o.voided =0
-AND o.person_id=t.patient_id 
-AND o.encounter_id=t.encounter_id 
-limit 1
-);
-
--- --------------------------- estimated delivery date ----------------------------------------------------
-
-UPDATE salud_mental_encountero t 
-SET t.estimated_delivery_date  = (
-SELECT cast(DATE_ADD(value_datetime, interval 280 day) as date)
-FROM obs o
-WHERE concept_id=@lastperioddate
-and o.voided =0
-AND o.person_id=t.patient_id 
--- AND o.encounter_id=t.encounter_id 
-order by encounter_id desc
-limit 1
-)
-WHERE t.prenatal_care is true;
+update salud_mental_encountero set grief = FALSE, adaptive_disorders = FALSE;
+update salud_mental_encountero e inner join temp_obs o on e.encounter_id = o.encounter_id set e.grief = TRUE, adaptive_disorders = TRUE
+where o.value_coded in (@grief);
 
 -- ------------------------------------ Diagnosis -------------------------------------------------------------------------------------------------------------------------------------------
-create or replace view diagnosis_pre_data as 
-	      select  
-			distinct person_id, encounter_id,
-			o.obs_group_id,o.voided,o.concept_id,
-			case when concept_name(value_coded,'en') is not null then concept_name(value_coded,'es') else value_text end as diagnosis,
-			case when concept_name(value_coded,'en') is not null then concept_name(value_coded,'en') else value_text end as diagnosis_en
-			from obs o 
-			left outer join concept_name cn 
-			on o.concept_id = cn.concept_id 
-			where locale='en'
-	      AND o.concept_id =concept_from_mapping('PIH',3064)
-	      AND o.voided = 0; 
-
-CREATE OR REPLACE VIEW diagnosis_data as 
-select person_id, encounter_id, 
-		group_concat(diagnosis SEPARATOR '; ') as diagnosis,
-		group_concat(diagnosis_en) as diagnosis_en
-		from diagnosis_pre_data
-group by person_id, encounter_id;
-
-UPDATE salud_mental_encountero t 
-SET t.diagnosis = (
-SELECT diagnosis
-FROM diagnosis_data dd
-WHERE dd.person_id=t.patient_id 
-AND dd.encounter_id=t.encounter_id 
-limit 1
-);
-
-
-UPDATE salud_mental_encountero t 
-SET t.diagnosis_en = (
-SELECT diagnosis_en
-FROM diagnosis_data dd
-WHERE dd.person_id=t.patient_id 
-AND dd.encounter_id=t.encounter_id 
-limit 1
-);
-
-UPDATE salud_mental_encountero t 
-SET t.primary_diagnosis  = (
-SELECT diagnosis
-FROM diagnosis_pre_data dp
-WHERE dp.person_id=t.patient_id 
-AND dp.encounter_id=t.encounter_id 
-limit 1
-);
 
--- ---- Ascending Order ------------------------------------------
```

Comment:
We should remove all of this ordering stuff I believe, in favor of doing this on the SQL Server side after the data has been loaded.  I removed it with that assumption.  @dmdesimone can you confirm?